### ISSUE TITLE
semi: [2, "never"] not erroring when semi-colon is after a static property

### ISSUE THREAD

**Issue by jeffshaver** [eslint/eslint#5366]
ESLint Version: 2.1.0

If I have `semi: [2, "never"]` set, I would imagine this would cause an error:

```javascript

export class MyClass extends Component {
  static propTypes = {
  }; // <-- expected this to give an error

  render () {
    // code
  }
}
```

Currently, I am getting 0 output from ESLint about this.

* [Actual source code](https://github.com/jeffshaver/safe-framework/blob/scatter-plot-component/src/DefaultScatterPlot/DefaultScatterPlot.jsx#L13)

* [ESLint Config](https://github.com/jeffshaver/safe-framework/blob/scatter-plot-component/.eslintrc)

**Comment by nzakas** [comment]
This is because you are using babel-eslint and an experimental ES feature. Please follow up with babel-eslint.

**Comment by jeffshaver** [comment]
Here is the babel-eslint issue:  babel/babel-eslint#260
